editoast/src/schema/rolling_stock/mod.rs

```diff
@@ -180,6 +180,15 @@ impl<'de> Deserialize<'de> for EffortCurve {
             ));
         }
 
+        let mut seen_coordinates: Vec<(f64, f64)> = Vec::new();
+        for (&effort, &speed) in inner.max_efforts.iter().zip(inner.speeds.iter()) {
+            let coord = (effort, speed);
+            if seen_coordinates.contains(&coord) {
+                return Err(serde::de::Error::custom("The effort curve is invalid: it contains duplicate (effort, speed) coordinates."));
+            }
+            seen_coordinates.push(coord);
+        }
```

Comment:
This is not the right behavior. We only have to check that the list of the speed has strictly increasing values.
Note: We can also check that speeds and efforts values are `>= 0`.